tests/tests.py

```diff
@@ -399,6 +400,49 @@ def test_index_can_hide_all_emails(self, mock_settings):
         self.assertNotIn('alpha@alpha.alpha', response.content)
         self.assertNotIn('bravo@bravo.bravo', response.content)
 
+    def test_check_requires_permission(self):
+        self.client.login(username='superuser', password='superuser')
+        response = self.client.get(reverse('invite:check'))
+        self.assertTrue(response.status_code == 200)
+        self.client.logout()
+        self.client.login(username='normal', password='normal')
+        response = self.client.get(reverse('invite:check'))
+        self.assertTrue(response.status_code == 403)
+
+    def test_check_returns_true_when_taken(self):
+        self.client.login(username='superuser', password='superuser')
+        taken = (
+            {'username': self.normal_user.username},
+            {'username': self.supertwo.username},
+            {'email': self.normal_user.email},
+            {'email': self.supertwo.email},
+        )
+        for each in taken:
+            response = self.client.get(reverse('invite:check'), each)
+            self.assertTrue(json.loads(response.content)['taken'] is True)
+
+    def test_check_returns_false_when_not_taken(self):
+        self.client.login(username='superuser', password='superuser')
+        not_taken = (
+            {'username': 'doesnotexist'},
+            {'username': ''},
+            {'email': 'nothing@nothing.nothing'},
+            {'email': 'test@test.test'},
+        )
+        for each in not_taken:
+            response = self.client.get(reverse('invite:check'), each)
+            self.assertTrue(json.loads(response.content)['taken'] is False)
+
+    def test_check_returns_false_with_no_args(self):
+        self.client.login(username='superuser', password='superuser')
+        response = self.client.get(reverse('invite:check'))
+        self.assertTrue(json.loads(response.content)['taken'] is False)
+
+    def test_check_returns_false_with_unrecognized_args(self):
+        self.client.login(username='superuser', password='superuser')
+        response = self.client.get(reverse('invite:check'), {'rand': 'rand'})
+        self.assertTrue(json.loads(response.content)['taken'] is False)
```

Comment:
I am not sure if this test is really necessary as it seems like it would fall into the same case as the above test. But if we need to keep it, you could replace with an `assertFalse` equivalent.

Reply:
Yeah, I think you're right, this doesn't hit any piece of code that the previous test didn't. I'll get rid of it.